src/smc-webapp/project_store.coffee

```diff
@@ -1200,15 +1200,17 @@ class ProjectActions extends Actions
                 if target[target.length-1] == '/' or full_path == ''
                     @open_directory(parent_path)
                 else
-                    @fetch_directory_listing(parent_path)
+                    @fetch_directory_listing(parent_path, true, true)
                     @get_store().wait
-                        until   : (s) => s.directory_listings.get(parent_path)
+                        until   : (s) =>
+                            listing = s.directory_listings.get(parent_path)
+                            return listing?.find (val) => val.get('name') == last
                         timeout : 3
-                        cb      : (err, listing) =>
+                        cb      : (err, item) =>
                             if err
-                                alert_message(type:'error', message:'Failed to open link')
+                                console.log err
```

Comment:
This console.log probably shouldn't be here.  If you really have an error that the user should know about, it shouldn't be hidden in the console.   Instead give an alert message that is helpful, e.g., include the path in it.

It seems like you changed the code to basically silently ignore an error and do something different (show a directory).